src/balance/balance.service.spec.ts

```diff
@@ -60,7 +72,7 @@ describe('BalanceService', () => {
     });
 
     it('throws an error if the user does not exist', async () => {
-      mockUserRepository.findById.mockResolvedValueOnce(null);
+      mockUserRepository.findById.mockImplementationOnce(() => null);
```

Comment:
Consider checking if moving back to `mockResolvedValueOnce` can still be used.  This was changed due to something else, I think.